dynosaur_derive/src/expand.rs

```diff
@@ -201,7 +274,15 @@ fn expand_arrow_ret_ty(sig: &Signature) -> (RArrow, TokenStream) {
         }
         (false, ReturnType::Type(arrow, ret)) => {
             if let Type::ImplTrait(TypeImplTrait { bounds, .. }) = &**ret {
-                return (*arrow, quote!(#bounds));
+                let mut ret_bounds: Punctuated<&TypeParamBound, Token![+]> = Punctuated::new();
+
+                for bound in bounds {
```

Comment:
Why filter out lifetimes here?

Reply:
This is the return type, we filter and at the end use the `'dynosaur` one